EngineLayer/CompactToProteinMatchEngine/SequencesToActualProteinPeptidesEngine.cs

```diff
@@ -48,7 +48,15 @@ protected static void ResolveAmbiguities(Dictionary<CompactPeptideBase, HashSet<
             foreach (CompactPeptide key in keys)
             {
                 HashSet<PeptideWithSetModifications> value = compactPeptideToProteinPeptideMatching[key];
-                compactPeptideToProteinPeptideMatching[key] = new HashSet<PeptideWithSetModifications> { value.FirstOrDefault(b => !b.Protein.IsDecoy) ?? value.First() };
+                HashSet<PeptideWithSetModifications> resolvedValues = new HashSet<PeptideWithSetModifications>();
```

Comment:
revert

Reply:
If i revert this then parsimony will fail